demos/ag-alert-keyvault-advisor/azuredeploy.json

```diff
@@ -0,0 +1,64 @@
+{
+   "$schema":"https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+   "contentVersion":"1.0.0.0",
+   "parameters":{
+      "alertName":{
+         "type":"String"
+      },
+      "applicationGatewayResourceGroup":{
+         "type":"String",
+         "metadata":{
+            "description":"Resource Group example /subscriptions/subs-id/resourceGroups/rg-name"
+         }
+      },
+      "actionGroupResourceID":{
+         "type":"String",
+         "metadata":{
+            "description":"Action Group ResourceID example /subscriptions/subs-id/resourceGroups/rg-name/providers/microsoft.insights/actiongroups/action-group-name"
+         }
+      }
+   },
+   "variables":{
+      
+   },
+   "resources":[
+      {
+         "type":"microsoft.insights/activityLogAlerts",
+         "apiVersion":"2020-10-01",
+         "name":"[parameters('alertName')]",
+         "location":"Global",
+         "properties":{
+            "scopes":[
+               "[parameters('applicationGatewayResourceGroup')]"
+            ],
+            "condition":{
+               "allOf":[
+                  {
+                     "field":"properties.recommendationType",
+                     "equals":"3467464b-955a-4caf-95e5-547344ba0281"

```

Comment:
how does the user know what this guid indicates?

Reply:
The user doesn't need to know the GUID of the alert type. The description of this alert "Setup alert for misconfigured Key Vault for your Azure Application Gateway" signifies what the alert is meant for and will be deplyed for that precise requirement.

Reply:
Where do I find the list of guids/alerts?

Reply:
https://docs.microsoft.com/en-us/rest/api/advisor/recommendation-metadata/list This should help

However, related to our product there are barely 3-4 alerts, and of those we want to nudge customers for this specific critical alert only.

Reply:
Is there something we can put into the readme so customers can learn more?

Reply:
That is outside of scope of Application Gateway product. We have only one alert that is focusing on our product's scenario